OpenRA.Mods.Common/Traits/AutoCarryall.cs

```diff
@@ -102,17 +158,66 @@ void FindCarryableForTransport(Actor self)
 			}
 		}
 
+		void IResolveOrder.ResolveOrder(Actor self, Order order)
+		{
+			if (order.OrderString == "DeliverUnit")
+			{
+				var cell = self.World.Map.Clamp(self.World.Map.CellContaining(order.Target.CenterPosition));
+				if (!AircraftInfo.MoveIntoShroud && !self.Owner.Shroud.IsExplored(cell))
+					return;
+
+				if (underAutoCommand)
+				{
+					underAutoCommand = false;
+					self.CancelActivity();
+				}
+
+				busy = true;
+				self.QueueActivity(order.Queued, new DeliverUnit(self, order.Target, Info.DropRange, Info.TargetLineColor));
+				self.ShowTargetLines();
+			}
+			else if (order.OrderString == "Unload")
+			{
+				if (!order.Queued && !CanUnload())
+					return;
+
+				if (underAutoCommand)
+				{
+					underAutoCommand = false;
+					self.CancelActivity();
+				}
+
+				busy = true;
+				self.QueueActivity(order.Queued, new DeliverUnit(self, Info.DropRange, Info.TargetLineColor));
+			}
+			else if (order.OrderString == "PickupUnit")
+			{
+				if (order.Target.Type != TargetType.Actor)
+					return;
+
+				if (underAutoCommand)
+				{
+					underAutoCommand = false;
+					self.CancelActivity();
+				}
+
+				busy = true;
+				self.QueueActivity(order.Queued, new PickupUnit(self, order.Target.Actor, Info.BeforeLoadDelay, Info.TargetLineColor));
+				self.ShowTargetLines();
+			}
+		}
+
 		sealed class FerryUnit : Activity
 		{
 			readonly Actor cargo;
-			readonly Carryable carryable;
-			readonly Carryall carryall;
+			readonly AutoCarryable carryable;
+			readonly AutoCarryall carryall;
 
```

Comment:
As `AutoCarryall` is now selectable, `FerryUnit` activity should show target lines

```suggestion
public override IEnumerable<TargetLineNode> TargetLineNodes(Actor self)
{
	if (ChildActivity != null)
		foreach (var n in ChildActivity.TargetLineNodes(self))
			yield return n;
}
```

Reply:
fixed